File: backend/app/schemas/analysis.py
"""CV analysis schemas."""

from __future__ import annotations

from datetime import datetime
from typing import Any

from pydantic import BaseModel, Field

from app.schemas.common import ORMModel


class AnalyzeRequest(BaseModel):
    cv_id: int
    job_description_id: int | None = None


class AnalysisResult(BaseModel):
    """Shape the AI service is asked to return for an analysis."""

    score: int | None = Field(default=None, ge=0, le=100)
    strengths: list[str] = Field(default_factory=list)
    weaknesses: list[str] = Field(default_factory=list)
    missing: list[str] = Field(default_factory=list)
    recommendations: list[str] = Field(default_factory=list)


class AnalysisResponse(ORMModel):
    id: int
    cv_id: int | None = None
    cv_version_number: int | None = None
    score: int | None = None
    results: dict[str, Any] = Field(default_factory=dict)
    recommendations: list[Any] = Field(default_factory=list)
    created_at: datetime

File: frontend/src/types/analysis.ts
// Mirrors backend app/schemas/analysis.py.

export interface AnalysisResult {
  score?: number | null
  strengths: string[]
  weaknesses: string[]
  missing: string[]
  recommendations: string[]
}

export interface AnalysisResponse {
  id: number
  cv_id: number | null
  cv_version_number: number | null
  score: number | null
  results: Partial<AnalysisResult>
  recommendations: string[]
  created_at: string
}

export interface AnalyzeRequest {
  cv_id: number
  job_description_id?: number
}
